src/agent/src/cdh.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// Confidential Data Hub client wrapper.
+// Confidential Data Hub is a service running inside guest to provide resource related APIs.
+// https://github.com/confidential-containers/guest-components/tree/main/confidential-data-hub
+
+use anyhow::Result;
+use derivative::Derivative;
+use protocols::{
+    sealed_secret, sealed_secret_ttrpc_async, sealed_secret_ttrpc_async::SealedSecretServiceClient,
+};
+
+use crate::CDH_SOCKET_URI;
+
+// Nanoseconds
+const CDH_UNSEAL_TIMEOUT: i64 = 50 * 1000 * 1000 * 1000;
+const SEALED_SECRET_PREFIX: &str = "sealed.";
+
+#[derive(Derivative)]
+#[derivative(Clone, Debug)]
+pub struct CDHClient {
+    #[derivative(Debug = "ignore")]
+    sealed_secret_client: SealedSecretServiceClient,
+}
+
+impl CDHClient {
+    pub fn new() -> Result<Self> {
+        let client = ttrpc::asynchronous::Client::connect(CDH_SOCKET_URI)?;
+        let sealed_secret_client =
+            sealed_secret_ttrpc_async::SealedSecretServiceClient::new(client);
+
+        Ok(CDHClient {
+            sealed_secret_client,
+        })
+    }
+
+    pub async fn unseal_secret_async(&self, sealed_secret: &str) -> Result<Vec<u8>> {
+        let mut input = sealed_secret::UnsealSecretInput::new();
+        input.set_secret(sealed_secret.into());
+
+        let unsealed_secret = self
+            .sealed_secret_client
+            .unseal_secret(ttrpc::context::with_timeout(CDH_UNSEAL_TIMEOUT), &input)
+            .await?;
+        Ok(unsealed_secret.plaintext)
+    }
+
+    pub async fn unseal_env(&self, env: &str) -> Result<String> {
+        let (key, value) = env.split_once('=').unwrap();
+        if value.starts_with(SEALED_SECRET_PREFIX) {
+            let unsealed_value = self.unseal_secret_async(value).await?;
+            let unsealed_env = format!("{}={}", key, std::str::from_utf8(&unsealed_value)?);
```

Comment:
@fitzthum Does k8s spec envvar secrets' value to be UTF-8? or it's our convention?

Reply:
Great question. It's our convention. k8s just sees the secret as a string, but we know that it's actually a wrapped secret. utf-8 tells us how to get back to a string after we decrypt the secret. tbh we haven't documented our convention very well. In theory this should be handled by the [secret-cli](https://github.com/confidential-containers/guest-components/blob/main/confidential-data-hub/secret/src/bin/secret_cli.rs#L64) but we don't support vault secrets there yet. I will probably add that support soon. It also might be worth reworking the API of the CDH to return a string rather than bytes, but that is a different discussion.

Reply:
Got it. Thanks for the explanation!

Reply:
btw this is why this PR has the secret hard-coded. once we have the secret-cli updated, I will use that in the test instead.

Reply:
Cool!